charts/prometheus/templates/server/sts.yaml

```
@@ -194,6 +192,10 @@ spec:
           {{- if .Values.server.extraVolumeMounts }}
           {{ toYaml .Values.server.extraVolumeMounts | nindent 12 }}
           {{- end }}
+          {{- if .Values.server.containerSecurityContext }}
+          securityContext:
+            {{- toYaml .Values.server.containerSecurityContext | nindent 12 }}
+          {{- end }}
```

Why not using `with`?

```suggestion
          {{- with .Values.server.containerSecurityContext }}
          securityContext:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```

If somebody defines `.Values.server.containerSecurityContext` as `{}` then it shall output nothing.